### DISCUSSION TITLE
Wakes are getting deflected when the yaw offset is set to zero

### DISCUSSION THREAD

**Discussion by prith-gs** [NatLabRockies/floris#991] [General]
Hello, 

I have a simple case with three turbines in FLORIS 3.5. 
The turbines are spaced roughly 6 diameters apart and are in line with the flow. 

Wind speed = 8 m/s, Wind direction = 180 degrees (south) and the turbines are all in line with the inflow.

I'm using the Jensen Wake velocity model, Gaussian deflection model, SOS superposition model and with secondary steering, yaw added recovery and transverse velocities enabled.

Below, you can see my inputs. 
![image](https://github.com/user-attachments/assets/6d9ce199-d914-4b69-859d-1316c13f979d)

In the jensen.py file in floris/simulation/wake_velocity I added a line to print the deflection_field_i. In line 102, I have added the print command before the calculation begins. I have not modified anything else in the file.
I used the reshape function to change the array with size = [1,1,3,1,1] to size = [3] so that the array is easy to read when it's printed.

![image](https://github.com/user-attachments/assets/93fd3eb3-fb49-4288-b207-e2c375f32f76)

Now, I run two cases with the before mentioned three turbines.
Case 1: None of the turbines have any yaw offsets.
Case 2: The first turbine is yawed by 20 degrees

Below is the script I'm running.
Line 10: I'm loading the input file
Line 14: Case 1 with no yaw offsets
Line 19: Case 2 with the first turbine yawed by 20 degrees
![image](https://github.com/user-attachments/assets/511be0ca-7fe3-4834-b674-76ad4948e415)

and below is the output
![image](https://github.com/user-attachments/assets/3a65bd87-568a-4be5-bccf-b69847c6d4ad)

**Why does case 1 when the turbines are not yawed has a non-zero deflection field?**
-  I notice that this only happens when secondary steering, yaw added recovery and transverse velocities are enabled. If I disable them then the deflection field is all zeros for case 1.
- This is my understanding of secondary steering - "Secondary steering is defined as the wake deflection at a turbine in the wake of a yaw-misaligned turbine. Due to the skew introduced to the flow by yaw-misalignment, turbines downwind operate as if they are misaligned with the flow." - So, when the turbines are not yawed at all, enabling secondary steering should also not make a difference. The deflection field should be all zeros. 
 
Is my understanding of secondary steering wrong? or is this a bug?

I'm not sure if this happens in the later versions of FLORIS (> 3.5). Looking forward to your response.

**Comment by misi9170** [comment]
Hi @prith-gs ,

Thanks for your question! I've checked, and this does still occur in FLORIS v4.1. I believe that the reason for this small deflection even when turbines are aligned is that, when the `enable_secondary_steering` flag is `true`, there is a (small) effective yaw angle computed by the `wake_added_yaw` method in wake_deflection/gauss.py (this is [here in v3.6](https://github.com/NREL/floris/blob/v3.5/floris/simulation/wake_deflection/gauss.py#L249) or [here in the v4.1](https://github.com/NREL/floris/blob/v4.1/floris/core/wake_deflection/gauss.py#L235). As best I understand it, this is an artifact of including curled-wake model aspects into the Gaussian wake model, which is described in detail in Section 2 of [King et al. (2021)](https://wes.copernicus.org/articles/6/701/2021/wes-6-701-2021.html).

Cheers,
Misha

**Comment by prith-gs** [comment]
Thank you! 

Because of the rotation of the wind turbine the wakes downstream are always deflected even when an isolated turbine is aligned with the flow, right? In general, the low fidelity models do not account for this effect.

Can I say that this accounts for that effect in some way? What do you guys think?

**Comment by misi9170** [comment]
Hi @prith-gs , I spoke a bit further with @tonyinme about this. Yes, that's essentially right---the rotation of the upstream turbine causes a vortex, which results in a small lateral flow at the rotor. In FLORIS, this lateral flow is modeled by computing an "effective yaw angle" that deflects the wake slightly according to the deflection model. The calculation of the lateral flow component, as well as its representation as an effective yaw angle, are of course very approximate (since we compute the integral of the curl over the rotor disk using a sum over only a handful of grid points); moreover, the lateral velocity at the top and bottom of the rotor do not cancel out exactly because the ground is modeled using a mirror wake in this process (see red circled sentence below, which comes from King et al. (2021)).

I hope this helps---@tonyinme, please feel free to correct anything I've said wrong!

Misha
<img width="773" alt="Screenshot 2024-10-09 at 5 12 44 PM" src="https://github.com/user-attachments/assets/b5be4c71-8751-4314-a903-6123fff13507">

**Reply by tonyinme** [reply]
Sounds great, thanks @misi9170 . 

Let us know if you have any other questions.

**Comment by prith-gs** [comment]
Thanks so much! That cleared things for me. Appreciate the response. :)
